lib/src/relationship_generator.dart

```diff
@@ -0,0 +1,57 @@
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/element/element.dart';
+import 'package:analyzer/dart/element/type.dart';
+import 'package:scip_dart/src/gen/scip.pbserver.dart';
+import 'package:scip_dart/src/symbol_generator.dart';
+
+List<Relationship>? relationshipsFor(
+  Declaration node,
+  Element element,
+  SymbolGenerator symbolGenerator,
+) {
+  // Only classes have relationships. Mixins and enums do not support
+  // inheritance themselves and therefore do not have any relationships
+  // to define.
+  if (element is ClassElement) {
+    // All things inherit from Object, if they have one supertype
+    // there are no relationships
+    if (element.allSupertypes.length <= 1) return null;
+
+    return element.allSupertypes
+        .where((type) => !type.isDartCoreObject)
+        .map((type) => Relationship(
+              symbol: symbolGenerator.symbolFor(type.element),
+              isImplementation: true,
+            ))
```

Comment:
If a class has any supertypes, all of those supertypes (`allSupertypes` refers to _every_ super type, including super types of super types) are relationships